billing-report-agent/src/main/java/com/epam/pipeline/billingreportagent/service/impl/converter/AwsStoragePriceListLoader.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright 2017-2020 EPAM Systems, Inc. (https://www.epam.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.epam.pipeline.billingreportagent.service.impl.converter;
+
+import com.amazonaws.regions.Regions;
+import com.amazonaws.services.pricing.AWSPricing;
+import com.amazonaws.services.pricing.AWSPricingClientBuilder;
+import com.amazonaws.services.pricing.model.Filter;
+import com.amazonaws.services.pricing.model.GetProductsRequest;
+import com.amazonaws.services.pricing.model.GetProductsResult;
+import com.epam.pipeline.billingreportagent.model.billing.StoragePricing;
+import com.epam.pipeline.billingreportagent.model.pricing.AwsPriceDimensions;
+import com.epam.pipeline.billingreportagent.model.pricing.AwsPriceRate;
+import com.epam.pipeline.billingreportagent.model.pricing.AwsPricingCard;
+import com.epam.pipeline.entity.region.CloudProvider;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.math.MathContext;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Slf4j
+public class AwsStoragePriceListLoader implements StoragePriceListLoader {
+
+    private static final String AWS_PRICE_FORMAT_VERSION = "aws_v1";
+    private static final String LOCATION_KEY = "location";
+    private static final String TERM_MATCH_FILTER = "TERM_MATCH";
+    private static final String PRODUCT_FAMILY_KEY = "productFamily";
+    private static final String STORAGE = "Storage";
+    private static final String STORAGE_CLASS_KEY = "storageClass";
+    private static final String GENERAL_STORAGE = "General Purpose";
+    private static final String US_DOLLAR_CODE = "USD";
+
+    private final String awsStorageServiceName;
+    private final ObjectMapper mapper;
+
+    public AwsStoragePriceListLoader(final String awsStorageServiceName) {
+        this.awsStorageServiceName = awsStorageServiceName;
+        this.mapper = new ObjectMapper();
```

Comment:
To prevent some common errors, we can set property `FailOnUnknownProperties` to false for this mapper. In this case parsing won't fail if some new fields will be added to AWS model.

Reply:
Done